plugins/ConvertStacked/ConvertStacked.cpp

```diff
@@ -134,13 +134,41 @@ class ConvertFromStacked : public GenericVideoFilter
 {
 public:
 
-    ConvertFromStacked::ConvertFromStacked(PClip src, IScriptEnvironment* env) : GenericVideoFilter(src)
+    ConvertFromStacked::ConvertFromStacked(PClip src, int bits, IScriptEnvironment* env) : GenericVideoFilter(src)
     {
-        if (vi.IsYV12()) vi.pixel_type = VideoInfo::CS_YUV420P16;
-        else if (vi.IsYV16()) vi.pixel_type = VideoInfo::CS_YUV422P16;
-        else if (vi.IsYV24()) vi.pixel_type = VideoInfo::CS_YUV444P16;
-        else if (vi.IsY8()) vi.pixel_type = VideoInfo::CS_Y16;
-        else env->ThrowError("ConvertFromStacked: Input stacked clip must be YV12, YV16, YV24 or Y8");
+        if (bits == 10 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P10;
+        else if (bits == 10 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P10;
+        else if (bits == 10 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P10;
+        else if (bits == 10 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y10;
+        else if (bits == 12 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P12;
+        else if (bits == 12 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P12;
+        else if (bits == 12 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P12;
+        else if (bits == 12 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y12;
+        else if (bits == 14 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P14;
+        else if (bits == 14 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P14;
+        else if (bits == 14 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P14;
+        else if (bits == 14 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y14;
+        else if (bits == 16 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P16;
+        else if (bits == 16 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P16;
+        else if (bits == 16 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P16;
+        else if (bits == 16 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y16;
+        else env->ThrowError("ConvertStackedToNative: Input stacked clip must be YV12, YV16, YV24 or Y8");
```

Comment:
The results of what avs_bits_per_pixel reports:
>RGB24: 24
RGB32: 32
RGB48: 48
RGB64: 64
YUY2: 16
YV24: 03
YV12: 01
YV16: 02
YV411: 01
Y8: 08
Y16: 16
YUV444P16: 06

Compare this to AviSynth 2.6 (I took the hardcoded values away and used only avs_bits_per_pixel, as I was starting to suspect it wasn't really necessary either):
>YV24: 24
YV12: 12
YV16: 16
YV411: 12

So yeah, BitsPerPixel isn't reporting the right bit values for planar formats (it looks like it's reporting *byte* values instead).  Y8 and Y16 would be fine, and so would all the packed formats, but the rest aren't correct, and trying to undo that in libavformat is both A) not the right way of fixing it and B) would require if/else-ing it so that the planar and packed formats don't screw each other up.